src/backend/distributed/commands/extension.c

```diff
@@ -894,9 +905,10 @@ PreprocessAlterExtensionCitusStmtForCitusColumnar(Node *parseTree)
 		double newVersionNumber = GetExtensionVersionNumber(pstrdup(newVersion));
 
 		/*alter extension citus update to version >= 11.1-1, and no citus_columnar installed */
-		if (newVersionNumber >= 1110 && citusColumnarOid == InvalidOid)
+		if (newVersionNumber >= 1110 && citusColumnarOid == InvalidOid &&
+			(newVersionNumber < 1320 || HasAnyRelationsUsingOldColumnar()))
```

Comment:
Bit confused - we are checking for old extension `citus`, but what about relations using `citus_coumnar`?